src/spaces/addSpace.js

```diff
@@ -79,13 +79,33 @@ export async function loadAddSpaces() {
   let cardMap = {};
 
   for (let spaceData of spaces) {
-    let card = cardTemplate.content.cloneNode(true);
-    card.querySelector("img").src = browser.runtime.getURL(`/recipes/${spaceData.recipeId}/icon.svg`);
-    card.querySelector(".name").textContent = spaceData.title;
-    card.querySelector(".card")._spaceData = spaceData;
-
-    cardMap[spaceData.recipeId] = card.firstElementChild;
-    allCardsContainer.appendChild(card);
+    // Clone template content, then get the real element from the fragment
+    const frag = cardTemplate.content.cloneNode(true);
+    const cardEl = frag.firstElementChild;
+
+    // Populate visuals
+    cardEl.querySelector("img").src = browser.runtime.getURL(`/recipes/${spaceData.recipeId}/icon.svg`);
+    cardEl.querySelector(".name").textContent = spaceData.title;
+
+    // A11y and keyboard activation
+    cardEl.setAttribute("tabindex", "0");
+    cardEl.setAttribute("role", "button");
+    cardEl.setAttribute("aria-label", spaceData.title);
+    cardEl.addEventListener("keydown", (e) => {
```

Comment:
inline functions within a for loop will do weird things in javascript, it is possible the last card element would be clicked instead. Move the function out of the loop/function.